backend/kale/common/podutils.py

```diff
@@ -171,19 +174,19 @@ def _list_volumes(client, namespace, pod_name, container_name):
         #  result in an incomplete notebook snapshot.
         pvc = client.read_namespaced_persistent_volume_claim(pvc.claim_name,
                                                              namespace)
-        if pvc.spec.storage_class_name != ROK_CSI_STORAGE_CLASS:
-            msg = ("Found PVC with storage class '%s'. Only storage class '%s'"
-                   " is supported."
-                   % (pvc.spec.storage_class_name, ROK_CSI_STORAGE_CLASS))
+        if pvc.spec.storage_class_name != ROK_CSI_STORAGE_CLASS and pvc.spec.storage_class_name != ROOK_CEPHFS_CSI_STORAGE_CLASS:
```

Comment:
Likely worth making a list of supported storage classes rather than just adding a new case for each type. Hopefully we will have many types. With a list it can just check if the current storage class is in the list.

Reply:
Thanks for the reviews and tips. Removing this hardcoding is something I have been thinking about on the side for a bit already. A list is indeed an option but it is still hardcoded. A more elegant solution would be to check if the storage provider supports snapshots or to check if a snapshot class exists on the cluster. The issue this creates is knowing if RWO or RWX volumes are supported.

Reply:
With the latest commit I have added functions that check if the provisioner of the PVC has an accompanying VolumeSnapshotClass rather than hardcoding what storage classes are supported. This also checks if a VolumeSnapshotClass actually exists, as you could be using CephFS or CephRDB without having added the VolumeSnapshotClass in which case it also isn't possible to take volume snapshots.